src/include/denc.h

```diff
@@ -156,6 +158,22 @@ struct denc_traits {
     };
     WRITE_CLASS_DENC(foo_t)
 
+  Decode Construction
+  ===================
+
+  A class with a constructor like
+
+    class Foo {
+      ...
+      Foo(buffer::ptr::iterator& p, uint64_t features = 0);
+    }
+
+  Will have be deconde constructble. Instances of this class can be
+  efficiently decoded into containers.
+
+  Since decode-constructors are new, I require all of them to accept a
+  features argument, to avoid the combinatorial explosion.
```

Comment:
We need to be very careful with this.  We generally rely on being able to transition a class from unfeatured to featured and get compile errors on all encoding paths.  If callers for this pass in feature fields they aren't explicitly passed (e.g., from a features=0 arg) we may silently fill in bogus features.  (Hopefully we haven't already done this elsewhere!)